client/app/widgets/Comments/__tests__/Comments.spec.jsx

```diff
@@ -83,41 +82,34 @@ const component = <Comments formProps={formProps} />;
 
 describe('Comments', () => {
   beforeEach(() => {
-    axiosPostSpy = jest.spyOn(axios, 'post').mockImplementation(() => Promise.resolve({
+    jest.spyOn(axios, 'post').mockImplementation(() => Promise.resolve({
       data: { comment },
     }));
-    axiosDeleteSpy = jest.spyOn(axios, 'delete').mockImplementation(() => Promise.resolve({
+    jest.spyOn(axios, 'delete').mockImplementation(() => Promise.resolve({
       data: { id },
     }));
   });
 
   it('renders correctly', () => {
-    let wrapper = null;
-    expect(() => {
-      wrapper = mount(component);
-    }).not.toThrow();
-    expect(wrapper).not.toBeNull();
+    render(component);
+    expect(() => component).not.toThrow();
```

Comment:
Q: Is this needed? I'm wondering if this assertion is making a difference.

The previous 'enzyme' version of this test would `mount` the component and expect it not to throw. Now this appears to be equivalent to passing in an un-invoked function component:

```
const NewComponent = () => component;
expect(NewComponent).not.toThrow();
```

Does that work? 🤔 I'm not sure. I haven't used it this way before.

Reply:
TBH I am not sure about this. Based on the [documentation](https://testing-library.com/docs/react-testing-library/migrate-from-enzyme) `render` in RTL is more like `mount` than `shallow`. But I am not really familiar with jest also so I don't know the answer.

Reply:
>render in RTL is more like `mount`

I think you're right 👍  whereas enzyme's `shallow` avoids rendering the component's children, both `render` and `mount` do render the full tree.

Btw I tried a quick analog to see if that previous assertion on line 95 works:

```
  it('See if component throws', () => {
    const Foo = () => {
      throw new Error('oops'); // definitely throws an error
    };
    const foo = <Foo />;
    // render(foo); // commenting this line out for now
    const WrappedFoo = () => foo;
    expect(WrappedFoo).not.toThrow(); // similar scenario
  });
```

And the test still passes -- even though it should fail.

That suggests only passing in a component, without calling or rendering it, to an assertion does not `throw`. And if it can never fail, the test can't be reliable.

Now if you uncomment the `render` line though, the test _does_ fail. Because rendering it actually throws the error inside `Foo` and causes the test as a whole to fail. So previously, it looks like line 94 was doing line 95's work.